src/AutoHideDockContainer.cpp

```diff
@@ -0,0 +1,527 @@
+        {
+            addWidget(emptyWidget);
+            addWidget(d->DockArea);
+			break;
+        }
+    }
+
+	updateSize();
+	updateMask();
+
+	parent->registerAutoHideWidget(this);
+}
+
+//============================================================================
+void CAutoHideDockContainer::updateMask()
+{
+    setMask(d->getDockAreaWithSplitterRect());
+}
+
+//============================================================================
+void CAutoHideDockContainer::updateSize()
+{
+	const auto dockContainerParent = parentContainer();
+	const auto rootSplitter = dockContainerParent->rootSplitter();
+	const auto rect = rootSplitter->frameGeometry();
+	move(rect.topLeft());
+    resize(rect.width(), rect.height());
+}
+
+//============================================================================
+CAutoHideDockContainer::CAutoHideDockContainer(CDockWidget* DockWidget, CDockWidgetSideTab::SideTabBarArea area, CDockContainerWidget* parent) : 
+	CAutoHideDockContainer(DockWidget->dockManager(), area, parent)
+{
+	addDockWidget(DockWidget);
+	setDockSizeProportion(DockWidget->DefaultAutoHideDockProportion());
+}
+
+//============================================================================
+CAutoHideDockContainer::~CAutoHideDockContainer()
+{
+	ADS_PRINT("~CAutoHideDockContainer");
+
+	// Remove event filter in case there are any queued messages
+	qApp->removeEventFilter(this);
+
+	if (d->DockManager)
+	{
+		parentContainer()->removeAutoHideWidget(this);
+	}
+
+	delete d;
+}
+
+//============================================================================
+CSideTabBar* CAutoHideDockContainer::sideTabBar() const
+{
+	return parentContainer()->sideTabBar(d->SideTabBarArea);
+}
+
+//============================================================================
+CDockWidget* CAutoHideDockContainer::dockWidget() const
+{
+	return d->DockWidget;
+}
+
+//============================================================================
+void CAutoHideDockContainer::addDockWidget(CDockWidget* DockWidget)
+{
+	if (d->DockWidget)
+	{
+		// Remove the old dock widget at this area
+        d->DockArea->removeDockWidget(d->DockWidget);
+	}
+
+	d->DockWidget = DockWidget;
+    CDockAreaWidget* OldDockArea = DockWidget->dockAreaWidget();
+    if (OldDockArea)
+    {
+        OldDockArea->removeDockWidget(DockWidget);
+    }
+	d->DockArea->addDockWidget(DockWidget);
+	d->DockWidget->sideTabWidget()->updateOrientationAndSpacing(d->SideTabBarArea);
+
+	updateSize();
+	updateMask();
+}
+
+
+//============================================================================
+void CAutoHideDockContainer::setDockSizeProportion(float SplitterProportion)
+{
+	if (SplitterProportion < 0 || SplitterProportion > 1)
+	{
+		ADS_PRINT("SplitterProportion must be set between 0 and 1.");
+		return;
+	}
+
+	const auto dockSize = static_cast<int>(static_cast<float>(INT_MAX) * SplitterProportion);
+	const auto remainingSize = INT_MAX - dockSize;
+    switch (d->SideTabBarArea)
```

Comment:
This code causes the following warning when building:

Warning: enumeration value 'Top' not handled in switch [-Wswitch]

Reply:
The stylesheet seems to be broken somehow. If I run the application I can see many warnings like this:

`Warning: QPushButton(0x219ddbf89e0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddbf89e0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddca3f40, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddca3f00, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddcb4bb0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddcb49f0, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddbf89e0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddca3f40, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddca3f00, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddcb4bb0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddcb49f0, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddcb4270, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddcb4270, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219db0256e0, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1de0, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1fe0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1160, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddcb4270, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219db0256e0, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1de0, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1fe0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1160, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddce5160, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddce5160, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce7760, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce77a0, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce7c20, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce7a60, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddcb4270, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219db0256e0, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1de0, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1fe0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce1160, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddbf89e0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddca3f40, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddca3f00, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddcb4bb0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddcb49f0, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddce5160, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce7760, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce77a0, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce7c20, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddce7a60, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddee6820, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddee6820, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf97790, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf94f90, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf95150, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf95190, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddee6820, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf97790, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf94f90, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf95150, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf95190, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddfe4910, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddee6820, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddfe4910, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0285c0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0285c0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029a00, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029600, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029800, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddfbb5c0, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0285c0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029a00, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029600, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029800, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddfbb5c0, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddee6820, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddfe4910, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf97790, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf94f90, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf95150, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddf95190, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddfcbe70, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddfcbe70, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de059350, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0594d0, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0595d0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de059e90, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219ddfcbe70, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de059350, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0594d0, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0595d0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de059e90, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0285c0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029a00, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029600, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029800, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddfbb5c0, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de062160, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de062160, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029300, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de028b00, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0284c0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddc90b10, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de062160, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de029300, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de028b00, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0284c0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219ddc90b10, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0a2e60, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0285c0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0a2e60, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
DockWidget Label 5
Warning: QPushButton(0x219de064c20, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de062160, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de064c20, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de063220, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de063220, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9c90, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9650, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9a10, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9a50, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de063220, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9c90, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9650, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9a10, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0d9a50, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0eacc0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0eacc0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0ea100, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0e9300, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0e90c0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0e9500, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0eacc0, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0ea100, name="tabsMenuButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0e9300, name="detachGroupButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0e90c0, name="dockAreaAutoHideButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: ads::CTitleBarButton(0x219de0e9500, name="dockAreaCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de18df40, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de0a2e60, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))
Warning: QPushButton(0x219de18df40, name="tabCloseButton")  does not have a property named  "iconsize" ((null):0, (null))`

Reply:
> Warning: enumeration value 'Top' not handled in switch

Fixed, thanks!

> The stylesheet seems to be broken somehow

I didn't get these warnings but I suspect it's because the case was different before. See if the latest push fixes it for you :)

I also fixed one more bug where right click detach of an overlay dock would not cause it to cleanup the overlay properly.

Let me know if you find anything else!

Reply:
@SyarifFakhri Thank you for the fixes. I reimplemented the `CAutoHideDockContainer` based on a QFrame. That means the `CAutoHideDockContainer` is now a QFrame instead of a QSplitter and uses a custom `CResizeHandle` widget for resizing. So masking is not required anymore. I did this, because with the current QSplitter based implementation non opaque resizing was not possible because of the masked splitter.

Saving and restoring state for the new `CAutoHideDockContainer` is not completely implemented yet. If you like, you can fix the save and restore functionality for the new `CAutoHideDockContainer`.

 I also added a special `visual_studio_light.css` file for testing a visual studio like theme.

You will find all my changes in the **auto_hide_feature** branch.

Reply:
> I reimplemented the CAutoHideDockContainer based on a QFrame. 

Actually this was my initial implementation, but I changed it to `QSplitter` after you reported issues with Active X (I'm not entirely sure why changing to `QSplitter` fixed it though). Do you get any issues with Active X using your `QFrame` implementation?

> Saving and restoring state for the new CAutoHideDockContainer is not completely implemented yet. If you like, you can fix the save and restore functionality for the new CAutoHideDockContainer.

I will take a look and implement this :)

> I also added a special visual_studio_light.css file for testing a visual studio like theme.

Cool!

I will merge all your changes and fix anything I find, thanks!

Reply:
I implemented the save and restore function for the `CAutoHideDockContainer`. I did some initial tests. It seems to work but I see crashes and gliches when switching perspectives, so there seems to be some problems in the code outside `CAutoHideDockContainer`

Reply:
The following Settings file will help you to reproduce the crash when loading perspectives. Download the settings file, rename it to Settings.ini and then copy it to your ADS demo application.

[Settings.txt](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/files/9870183/Settings.txt)

In the software open the perspective **FloatingAutoHide6**, then open the perspective **Default** and the open the perspective **FloatingAutoHide6** again - crash!!.

Reply:
@SyarifFakhri  I fixed the cause of the crash. The problem is, the deletion of the SideTabBar will delete all tabs in this SideBar. If a perspective is loaded, FloatingWidgets may be deleted wich in turn delets its SideBars wich in turn deletes the tabs in this SideBar. The SideBar is not the owner of the tabs should outlive the deletion of the SideBar.

Reply:
Yep, the true owner is the dockwidget, it's merely borrowed by the side tab bar. Thanks for the fix!

Reply:
@githubuser0xFFFF I've merged all your changes and made some small ones:
1. Fixed a crash if you tried to add an auto hide container without it having a previous dockArea (e.g. if you added it programmatically on software start)
2. Related to above: I added a `setSize` function so that a user could resize the auto hide container on first add
3. Fixed the CMakeLists.txt (added ResizeHandle.h/cpp)
4. I added a resize margin so the user can't resize the auto hide container all the way to the end and get stuck

Thanks for your time and contributions so far! Please let me know if there's anything else to look into :)